public/js/apps/transaction/app.jsx

```diff
@@ -50,6 +51,10 @@ export default class TransactionApp extends Component {
     var state = this.state;
     var SignIn = this.props.SignIn;
     var Transaction = this.props.Transaction;
+    var product = products.get(state.productId);
+    var signInRequired = product.seller.kind !== 'donations';
+    console.log('sign-in required to transact product?', state.productId,
+                signInRequired ? 'Yes': 'No');
```

Comment:
I think of this still as a strawman type of patch. A few things we can do later:

- make sign-in optional for donations anyway
- maybe rely on the API to see if we need to sign-in: try to make a request and see if we get a 403 before we try to sign in